### PR TITLE
test | GIDK-3nfxzv-test | Aditi | new list design changes

### PR DESCRIPTION
**Pull request by aditi-tiwari17**
* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
https://app.clickup.com/t/3nfxzv

* **What is the current behavior?** (You can also link to an open issue here)

* **What is the new behavior (if this is a feature change)?**

* **Other information**:

### CODE REVIEW

**Review by shalinee-walkover — APPROVED** [review]

File: apps/web-giddh/src/app/daybook/daybook.component.html [hunk 1]
```diff
@@ -38,15 +38,16 @@
...
                             <input type="text" name="selectedDateRange"
                                 [value]="(selectedDateRangeUi) ? selectedDateRangeUi : ''"
                                 class="giddh-datepicker date-range-picker border-radius-2" />
                         </div>
-                        <div class="form-group cursor-pointer">
-                            <div class="advance-icon align-items-center"
+                        <div class="form-group cursor-pointer mobile-view float-right">
+                            <div class="advance-icon align-items-center mr-0 btn btn-primary"
                                 [tooltip]="commonLocaleData?.app_advance_search" (click)="onOpenAdvanceSearch()">
+                                <span class="pd-r05">Filter</span>
```
Comment (prateek-giddh) [inline apps/web-giddh/src/app/daybook/daybook.component.html:50]: Do add multi-lingual labels and avoid hard coded values

File: apps/web-giddh/src/app/daybook/daybook.component.html [hunk 2]
```diff
@@ -68,6 +69,13 @@
                                 (click)="exportDaybook()"
                                 [disabled]="!((daybookData$ | async)?.entries?.length > 0)">{{commonLocaleData?.app_export}}</button>
                         </div>
+                        <div class="form-group cursor-pointer desktop-view float-right">
+                                <div class="advance-icon align-items-center btn btn-primary"
+                                    [tooltip]="commonLocaleData?.app_advance_search" (click)="onOpenAdvanceSearch()">
+                                    <span class="pd-r05">Filter</span>
```
Comment (prateek-giddh) [inline apps/web-giddh/src/app/daybook/daybook.component.html:75]: Do add multi-lingual labels and avoid hard coded values

File: apps/web-giddh/src/app/invoice/preview/invoice.preview.component.html [hunk 3]
```diff
@@ -67,54 +103,23 @@
...
                         (click)="exportCsvDownload()" [ladda]="exportInvoiceRequestInProcess$ | async"
                         *ngIf="selectedVoucher === 'sales' && selectedInvoicesList.length && isExported"
                         data-spinner-size="20px">
                         <span class="icon-export-icon"></span>
                     </button>
                     <!-- E-way bill generation and export to excel for a single selected item in invoice ends sizes -->
                     <!-- Section to create New Sales Invoice, Cash Invoice, Credit Note and Debit Note -->
+                    <div class="advance-icon cursor-pointer align-items-center btn btn-primary"
+                    [tooltip]="commonLocaleData?.app_advance_search" (click)="toggleAdvanceSearchPopup(false)">
+                    <span class="pd-r05">Filter</span> <span class="icon-advance-filter"></span>
```
Comment (prateek-giddh) [inline apps/web-giddh/src/app/invoice/preview/invoice.preview.component.html:116]: Do add multi-lingual labels and avoid hard coded values

File: apps/web-giddh/src/app/invoice/proforma/proforma-list.component.html [hunk 4]
```diff
@@ -3,79 +3,81 @@
...
-                <!-- select app proforma -->
-                <div class="select-all-proformas-on-mobile-view mt-1 mr-r15" *ngIf="selectedItems.length > 0">
-                    <div class="checkbox-img mr-1">
-                        <img src="assets/images/unchecked.png" *ngIf="!allItemsSelected" (click)="toggleAllItems(true)">
-                        <img src="assets/images/checked.png" *ngIf="allItemsSelected" (click)="toggleAllItems(false)">
-                    </div>
-                    {{commonLocaleData?.app_select_all}}
+            <div class="advance-icon cursor-pointer align-items-center btn btn-primary" (click)="toggleAdvanceSearchPopup()"
+            [tooltip]="commonLocaleData?.app_advance_search">
+                    <span class="pd-r05">Filter</span>
```
Comment (prateek-giddh) [inline apps/web-giddh/src/app/invoice/proforma/proforma-list.component.html:78]: Do add multi-lingual labels and avoid hard coded values

File: apps/web-giddh/src/app/reports/components/advance-receipt-report/advance-receipt-report.component.html [hunk 5]
```diff
@@ -52,6 +48,11 @@
                     <button class="btn btn-filter mr-0" (click)="resetAdvanceSearch()"><span class="icon-cross"></span>
                         {{commonLocaleData?.app_clear_filter}}</button>
                 </div>
+                <div class="advance-icon cursor-pointer btn btn-primary mr-0" tooltip="{{commonLocaleData?.app_advance_search}}"
+                            (click)="openModal()">
+                            <span class="pd-r05">Filter</span>
```
Comment (prateek-giddh) [inline apps/web-giddh/src/app/reports/components/advance-receipt-report/advance-receipt-report.component.html:53]: Do add multi-lingual labels and avoid hard coded values

File: apps/web-giddh/src/app/reports/components/purchase-register-table-component/purchase.register.table.component.html [hunk 6]
```diff
@@ -2,8 +2,30 @@
     (localeData)="localeData = $event" (commonLocaleData)="commonLocaleData = $event">
     <table class="table basic giddh-table on-mobile-view">
         <thead>
+                        <tr class="register-table-th">
```
Comment (prateek-giddh) [inline apps/web-giddh/src/app/reports/components/purchase-register-table-component/purchase.register.table.component.html:5]: Please check the formatting of this section, appears to be more than 4 spaces in the diff

File: apps/web-giddh/src/assets/css/table.scss [hunk 7]
```diff
@@ -48,19 +58,24 @@
...
         border-bottom: none !important;
         white-space: nowrap !important;
+        color: $omiod;
+        border-top: 1px solid $dark-fog !important;
         &.text-break {
             white-space: normal !important;
         }
     }
+    tbody > tr > td:first-child{
+        color: #212528;
```
Comment (prateek-giddh) [inline apps/web-giddh/src/assets/css/table.scss:68]: Do use color var

**Review by prateek-giddh — CHANGES_REQUESTED** [review]

File: apps/web-giddh/src/app/contact/contact.component.html [hunk 8]
```diff
@@ -524,18 +573,16 @@ <h1 class="mr-b1">{{localeData?.type_message_body}}:</h1>
...
-                                </div>
-                                <div class="icon-pointer cursor-pointer icon-class-search">
+                                    {{localeData?.contacts}}
                                 </div>
                             </div>
 
                         </th>
 
-                        <th *ngIf="showFieldFilter.mobile">{{commonLocaleData?.app_mobile_no}}</th>
+                        <!--<th *ngIf="showFieldFilter.mobile">{{commonLocaleData?.app_mobile_no}}</th>-->
```
[the remark this answers is not in the document]
Comment (aditi-tiwari17) [reply apps/web-giddh/src/app/contact/contact.component.html]: done

File: apps/web-giddh/src/app/contact/contact.component.html [hunk 9]
```diff
@@ -133,97 +58,242 @@
...
-                                        (click)="openAddAndManage('vendor')">
-                                        {{localeData?.new_vendor}}
-                                    </button>
+                            </div>
+                            <div class="advance-icon cursor-pointer align-items-center btn btn-primary btn-group mr-0" *ngIf="activeTab !== 'aging'"
+                            #filterDropDownList="bs-dropdown" dropdown [autoClose]="false"
+                            (clickOutside)="hideListItems()" [tooltip]="localeData?.filter_toggle"
+                            [placement]="'right'">
+                                <div dropdownToggle>
+                                        <span class="pd-r05">Filter</span>
```
[the remark this answers is not in the document]
Comment (aditi-tiwari17) [reply apps/web-giddh/src/app/contact/contact.component.html]: done

File: apps/web-giddh/src/app/contact/contact.component.html [hunk 10]
```diff
@@ -756,13 +803,19 @@ <h1 class="mr-b1" style="margin-top:15px;">
...
-                            <td class="word-wrap" *ngIf="showFieldFilter.mobile "
-                                [attr.data-title]="commonLocaleData?.app_mobile_no">
+                                <div>
+                                    {{account.email || '-' }}
+                                </div>
+                                <div>
                                 {{account.mobileNo || '-'}}
+                                </div>
                             </td>
+                            <!--<td class="word-wrap" *ngIf="showFieldFilter.mobile "
```
[the remark this answers is not in the document]
Comment (aditi-tiwari17) [reply apps/web-giddh/src/app/contact/contact.component.html:813]: done

File: apps/web-giddh/src/app/daybook/daybook.component.html (hunk 1 above)
Comment (aditi-tiwari17) [reply apps/web-giddh/src/app/daybook/daybook.component.html]: done

File: apps/web-giddh/src/app/daybook/daybook.component.html (hunk 2 above)
Comment (aditi-tiwari17) [reply apps/web-giddh/src/app/daybook/daybook.component.html]: done

File: apps/web-giddh/src/app/invoice/preview/invoice.preview.component.html (hunk 3 above)
Comment (aditi-tiwari17) [reply apps/web-giddh/src/app/invoice/preview/invoice.preview.component.html]: done

File: apps/web-giddh/src/app/invoice/proforma/proforma-list.component.html (hunk 4 above)
Comment (aditi-tiwari17) [reply apps/web-giddh/src/app/invoice/proforma/proforma-list.component.html]: done

File: apps/web-giddh/src/app/reports/components/advance-receipt-report/advance-receipt-report.component.html (hunk 5 above)
Comment (aditi-tiwari17) [reply apps/web-giddh/src/app/reports/components/advance-receipt-report/advance-receipt-report.component.html]: done

File: apps/web-giddh/src/app/reports/components/purchase-register-table-component/purchase.register.table.component.html (hunk 6 above)
Comment (aditi-tiwari17) [reply apps/web-giddh/src/app/reports/components/purchase-register-table-component/purchase.register.table.component.html]: done

File: apps/web-giddh/src/assets/css/table.scss (hunk 7 above)
Comment (aditi-tiwari17) [reply apps/web-giddh/src/assets/css/table.scss]: done

**Review by prateek-giddh — APPROVED** [review]

**Review by prateek-giddh — COMMENTED** [review]
@aditi-tiwari17 Please do remove the code in contact.component.html:
```
<!--<td class="word-wrap" *ngIf="showFieldFilter.mobile "
    [attr.data-title]="commonLocaleData?.app_mobile_no">
    {{account.mobileNo || '-'}}
</td>-->
```

File: apps/web-giddh/src/app/contact/aging-report/aging-report.component.ts [hunk 11]
```diff
@@ -84,6 +85,20 @@ export class AgingReportComponent implements OnInit, OnDestroy {
     public getAgingReportRequestInProcess$: Observable<boolean>;
     /** Stores the current organization type */
     public currentOrganizationType: OrganizationType;
+    /** Stores the datepicker modal reference */
+    public modalRef: BsModalRef;
+    /** Stores the date field position in datepicker */
+    public dateFieldPosition: any = { x: 0, y: 0 };
+    /** Datepicker reference */
+    @ViewChild('datepickerTemplate', { static: true }) public datepickerTemplate: ElementRef;
+    public selectedRangeLabel: any = "";
```
Comment (prateek-giddh) [inline apps/web-giddh/src/app/contact/aging-report/aging-report.component.ts:94]: JS doc

File: apps/web-giddh/src/app/contact/aging-report/aging-report.component.ts [hunk 12]
```diff
@@ -334,6 +359,110 @@ export class AgingReportComponent implements OnInit, OnDestroy {
...
+     * Returns the placeholder for the current searched field
+     *
+     * @param {string} fieldName Field name for which placeholder is required
+     * @returns {string} Placeholder text
+     * @memberof AgingReportComponent
+     */
+    public getSearchFieldText(fieldName: string): string {
+        if (fieldName === 'name') {
+            return this.localeData?.search_name;
+        }
```
Comment (prateek-giddh) [inline apps/web-giddh/src/app/contact/aging-report/aging-report.component.ts:463]: default return type

File: apps/web-giddh/src/app/contact/contact.component.ts [hunk 13]
```diff
@@ -1451,4 +1463,55 @@ export class ContactComponent implements OnInit, OnDestroy {
...
+    }
+
+    /**
+     * Toogles the search field
+     *
+     * @param {string} fieldName Field name to toggle
+     * @param {*} el Element reference for focusing
+     * @memberof ContactComponent
+     */
+    public toggleSearch(fieldName: string, el: any) {
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/contact/contact.component.ts:1496]: add return type

File: apps/web-giddh/src/app/reports/components/advance-receipt-report/advance-receipt-report.component.html [hunk 14]
```diff
@@ -52,6 +48,11 @@
                     <button class="btn btn-filter mr-0" (click)="resetAdvanceSearch()"><span class="icon-cross"></span>
                         {{commonLocaleData?.app_clear_filter}}</button>
                 </div>
+                <div class="advance-icon cursor-pointer btn btn-primary mr-0" tooltip="{{commonLocaleData?.app_advance_search}}"
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/reports/components/advance-receipt-report/advance-receipt-report.component.html:51]: replace interpolation with []

File: apps/web-giddh/src/app/services/general.service.ts [hunk 15]
```diff
@@ -851,4 +851,18 @@ export class GeneralService {
...
+
+    /**
+     * Determines if an element is child element to another element
+     *
+     * @param {*} child Element received as child
+     * @param {*} parent Element received as parent
+     * @return {boolean} True, if element is child of another element
+     * @memberof GeneralService
+     */
+    public childOf(c, p): boolean {
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/services/general.service.ts:863]: don't use single character variable

**Pull request merged**